### PR TITLE
Module-level RNG instantiation for quantum_info/operators/random

### PR DESCRIPTION
**Pull request by jkelso662**
### Summary
Create module wide default_rng to increase speed when initialising random operators.

### Details and comments
Fixes https://github.com/Qiskit/qiskit-terra/issues/4339

### CODE REVIEW

**Review by kdk — APPROVED** [review]
Thanks @jkelso662 , this looks good!

**Comment by kdk** [comment]
Before we can merge, can you update your branch to the current master? (Or you can allow edits from maintainers from your branch and then our CI can merge.)

**Comment by scottkelso** [comment]
Thanks @kdk. I have kicked off a GitHub 'Update Branch' from here so waiting to see if that works. Failing that I think I have changed my settings to allow others to edit but if not I have also invited you as a collaborator.

**Pull request merged**
